src/lasuite/malware_detection/backends/jcop.py

```diff
@@ -0,0 +1,222 @@
+    """Task starting analysis process for a file."""
+    if not file_hash:
+        file = default_storage.open(file_path, "rb")
+        file_hash = hashlib.file_digest(file, "sha256").hexdigest()
+        file.close()
+
+    callback = import_string(getattr(settings, setting_name("MALWARE_DETECTION_CALLBACK")))
+
+    api_key = getattr(settings, setting_name("JCOP_API_KEY"))
+    base_url = getattr(settings, setting_name("JCOP_BASE_URL"))
+    result_timeout = getattr(settings, setting_name("JCOP_RESULT_TIMEOUT"), 30)
+
+    try:
+        # try if the file as already been tested
+        response = requests.get(
+            f"{base_url}/results/{file_hash}",
+            headers={
+                "X-Auth-Token": api_key,
+                "Accept": "application/json",
+            },
+            timeout=result_timeout,
+        )
+    except requests.exceptions.RequestException as exc:
+        logger.error("Error getting cache result for file %s: %s", file_path, exc)
+        if self.request.retries >= self.max_retries:
+            callback(
+                file_path,
+                ReportStatus.UNKNOWN,
+                error_info={
+                    "error": "Max retries triggering new analysis exceeded",
+                    "error_code": 5000,
+                }
+                ** kwargs,
+            )
+        raise self.retry(exc=exc) from exc
+
+    if response.status_code == HTTPStatus.NOT_FOUND:
+        # start a new analysis
+        trigger_new_analysis.delay(file_path, **kwargs)
+        return
+
+    if response.status_code == HTTPStatus.UNAUTHORIZED:
+        callback(
+            file_path,
+            ReportStatus.UNSAFE,
+            error_info={
+                "error": "Invalid API key",
+                "error_code": response.status_code,
+            },
+            **kwargs,
+        )
+        raise MalwareDetectionInvalidAuthenticationError()
+
+    if response.status_code == HTTPStatus.OK:
+        content = response.json()
+        if not content["done"]:
+            # the analysis is not done yet, retry later
+            self.retry()
+            return
+
+        if content.get("error_code"):
+            status = ReportStatus.UNSAFE if content["is_malware"] else ReportStatus.UNKNOWN
+            callback(
+                file_path,
+                status,
+                error_info={
+                    "error": content["error"],
+                    "error_code": content["error_code"],
+                },
+                **kwargs,
+            )
+            return
+
+        if not content.get("is_malware"):
+            callback(file_path, ReportStatus.SAFE, error_info={}, **kwargs)
+            return
+
+    # Any other case, call the callback with an unknown error
+    callback(
+        file_path,
+        ReportStatus.UNKNOWN,
+        error_info={
+            "error": "Unknown treatment",
+            "error_code": response.status_code,
+        },
+        **kwargs,
+    )
```

Comment:
Don't you want to make an early return on this, like

```python
try:
    response.raise_for_status()
except ...

try:
    content = response.json()
except JSONDecodeError:  # new catch, just in case
     ....

if not content["done"]: ....
```

Reply:
Not sure it does what I want. I want to execute the callback in case No cases are matched.